elephant/joint_isi_dithering_class.py

```diff
@@ -0,0 +1,438 @@
+"""
+Module to generate surrogates of spike trains by using the joint-ISI dithering.
+Features are provided to separate the preprocessing from the main process.
+
+Original implementation by: Peter Bouss [p.bouss@fz-juelich.de]
+:copyright: Copyright 2019 by the Elephant team, see `doc/authors.rst`.
+:license: Modified BSD, see LICENSE.txt for details.
+"""
+
+import numpy as np
+import quantities as pq
+import neo
+
+import elephant.spike_train_surrogates as surr
+import elephant.statistics as stats
+from scipy.ndimage import gaussian_filter
+
+
+class JointISISpace:
+    """
+    The class :class:`Joint_ISI_Space` is implemented for Joint-ISI dithering
+    as a continuation of the ideas of Louis et al. (2010) and Gerstein (2004).
+
+    When creating an class instance all necessary preprocessing steps are done,
+    to use the method dithering().
+
+    To make the dithering-procedure stable, the spiketrain needs to pass two
+    thresholds, that the Joint-ISI dithering is applied, if not uniform
+    dithering is used.
+
+    Attributes
```

Comment:
All these attributes are "public", is this intentional, or should these rather be the arguments of the class constructor?

Reply:
This was intentional. My idea was to allow, changed this parameters afterwards. Which should allow to investigate easier all parameters. But maybe it is too much and not in all case useful. Since anyway, for most of them parts of the preprocessing needs to be redone.

Reply:
But effectively, of course it would be better to define update functions, where only the part of the preprocessing is done again, which is needed....
I will try to work it in the next days.